tools/dm-shard-merge-scenario.md

```diff
@@ -0,0 +1,273 @@
+---
+title: Data Migration Shard Merge Scenario
+summary: Learn how to use Data Migration to synchronize data in the shard merge scenario.
+category: tools
+---
+
+# Data Migration Shard Merge Scenario
+
+This document shows how to use Data Migration (DM) in the shard merge scenario where the sharded schemas and sharded tables data of three upstream MySQL instances need to be synchronized to a downstream TiDB cluster.
+
+## Upstream instances
+
+Assume that the upstream schemas are as follows:
+
+- Instance 1
+
+    | Schema | Tables|
+    |:------|:------|
+    | user  | information, log_north, log_bak |
+    | store_01 | sale_01, sale_02 |
+    | store_02 | sale_01, sale_02 |
+
+- Instance 2
+
+    | Schema | Tables|
+    |:------|:------|
+    | user  | information, log_east, log_bak |
+    | store_01 | sale_01, sale_02 |
+    | store_02 | sale_01, sale_02 |
+
+- Instance 3
+
+    | Schema | Tables|
+    |:------|:------|
+    | user  | information, log_south, log_bak |
+    | store_01 | sale_01, sale_02 |
+    | store_02 | sale_01, sale_02 |
+
+## Synchronization requirements
+
+1. Merge the `user`.`information` table of three upstream instances to the downstream `user`.`information` table in TiDB.
+2. Merge the `user`.`log_{north|south|east}` table of three upstream instances to the downstream `user`.`log_{north|south|east}` table in TiDB.
+3. Merge the `store_{01|02}`.`sale_{01|02}` table of three upstream instances to the downstream `store`.`sale` table in TiDB.
+4. Filter out all the deletion operations in the `user`.`log_{north|south|east}` table of three upstream instances.
+5. Filter out all the deletion operations in the `user`.`information` table of three upstream instances.
+6. Filter out all the deletion operations in the `store_{01|02}`.`sale_{01|02}` table of three upstream instances.
+7. Filter out the `user`.`log_bak` table of three upstream instances.
+8. Because the `store_{01|02}`.`sale_{01|02}` tables have auto-increment primary keys of the bigint type, the conflict occurs when these tables are merged into TiDB. So you need to modify the auto-increment primary keys to avoid the conflict.
+
+## Downstream instances
+
+Assume that the downstream schema after synchronization is as follows:
+
+| Schema | Tables |
+|:------|:------|
+| user | information, log_north, log_east, log_south|
+| store | sale |
+
+## Synchronization solution
+
+- To satisfy the synchronization Requirements #1 and #2, configure the [table routing rule](../tools/dm-data-synchronization-features.md#table-routing) as follows:
```

Comment:
is the link correct? and also more links below.

Reply:
@csuzhangxc Yes, they are correct. That file is added to docs in #839, which was merged already. I'll also check these links after this file shows on the official website.